src/math/lp/lar_solver.cpp

```diff
@@ -1709,12 +1716,55 @@ bool lar_solver::bound_is_integer_for_integer_column(unsigned j, const mpq & rig
     return right_side.is_int();
 }
 
+constraint_index lar_solver::add_var_bound_check_on_equal(var_index j, lconstraint_kind kind, const mpq & right_side, var_index& equal_var) {
+    constraint_index ci = mk_var_bound(j, kind, right_side);
+    activate_check_on_equal(ci, equal_var);
+    return ci;
+}
+
 constraint_index lar_solver::add_var_bound(var_index j, lconstraint_kind kind, const mpq & right_side) {
     constraint_index ci = mk_var_bound(j, kind, right_side);
     activate(ci);
     return ci;
 }
 
+void lar_solver::remove_non_fixed_from_fixed_var_table() {
+    vector<value_sort_pair> to_remove;
+    for (const auto& p : m_fixed_var_table) {
+        unsigned j = p.m_value;
+        if (j >= column_count() || column_is_fixed(j) == false)
```

Comment:
writing column_is_fixed(j) == false instead of !column_is_fixed(j) is not as bad as writing column_is_fixed(j) == true, but close.